### ISSUE TITLE
[receiver/kafka] Flaky TestIntegration

### ISSUE THREAD

**Issue by dmitryax** [open-telemetry/opentelemetry-collector-contrib#23267]
Seen in https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5230434021/jobs/9443984920?pr=23239

```
--- FAIL: TestIntegration (78.01s)
    scraperint.go:121: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/coreinternal/scraperinttest/scraperint.go:121
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/kafkametricsreceiver/integration_test.go:78
        	Error:      	Condition never satisfied
        	Test:       	TestIntegration
    scraperint.go:103: resource "map[]": scope "otelcol/kafkametricsreceiver": metric "kafka.brokers": datapoint "map[]": int value doesn't match expected: 1, actual: 0
    scraperint.go:109: full log:
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
        Error scraping metrics
    scraperint.go:117: latest result:
        resourceMetrics:
          - resource: {}
            scopeMetrics:
              - metrics:
                  - description: Number of brokers in the cluster.
                    gauge:
                      dataPoints:
                        - asInt: "0"
                          startTimeUnixNano: "1686407538913389877"
                          timeUnixNano: "1686407594914980439"
                    name: kafka.brokers
                    unit: '{brokers}'
                scope:
                  name: otelcol/kafkametricsreceiver
                  version: latest
FAIL
coverage: 29.1% of statements
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kafkametricsreceiver	78.068s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kafkametricsreceiver/internal/metadata	0.044s	coverage: 0.0% of statements [no tests to run]
FAIL
make[1]: *** [../../Makefile.Common:114: do-integration-tests-with-cover] Error 1
make: *** [Makefile:172: receiver/kafkametricsreceiver] Error 2
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/kafkametricsreceiver'
Error: Process completed with exit code 2.
```

**Comment by MovieStoreGuy** [comment]
All yours @shivanshu1333

**Comment by shivanshu1333** [comment]
@dmitryax I've tried to reproduce this, I ran the mentioned integration test ~20 times from my IDE, and every time the test was passed successfully, atm I'm unsure how to reproduce, any guidance? Thanks!

**Comment by dmitryax** [comment]
@shivanshu1333 I don't think it's possible to reproduce it locally. We need to read the code and figure out how this can happen

**Comment by dmitryax** [comment]
One more https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5673757659/job/15376061714?pr=24576

**Comment by atoulme** [comment]
One more https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5727989379/job/15521613294?pr=24638

**Comment by dmitryax** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5863608379/job/15897435670?pr=24444

**Comment by crobert-1** [comment]
Frequency found by @atoulme: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6347302582/job/17242118534?pr=27267

**Comment by MovieStoreGuy** [comment]
How are you going with this @shivanshu1333 ?

**Comment by songy23** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6813721514/job/18528980784?pr=29084

**Comment by djaglowski** [comment]
Closed by #30218
